doc/development/guide/architecture.rst

```diff
@@ -0,0 +1,204 @@
+contains basic information about the operator (e.g. number of parameters,
+number of wires it acts on, etc.) and further convenience methods (e.g.
+:attr:`~.Operator.matrix`, :attr:`~.Operator.eigvals`.
+
+Two important subclasses of the ``Operator`` class are:
+
+* the :class:`~.Operation` class representing quantum gates and
+* the :class:`~.Observable` representing quantum observables specified for
+  measurement.
+
+Together with ``Operator``, these classes serve as base classes for quantum
+operators.
+
+Certain operators can serve as both quantum gates and observables (e.g.
+:class:`~.PauliZ`, :class:`~.PauliX`, etc.). Such classes inherit from both
+``Operation`` and ``Observable`` classes.
+
+Quantum operator instances, in general, are used to build quantum functions
+which are used within ``QNode`` instances. Users can define such quantum
+functions by creating regular Python functions and incorporating ``Operator``
+instances.
+
+The following is an example of this using the :func:`~.qnode` decorator and a
+valid pre-defined device (``dev``).
+
+.. code-block:: python
+
+    @qml.qnode(dev)
+    def circuit():
+        qml.PauliX(0)
+        return qml.expval(qml.PauliZ(0))
+
+This syntax of PennyLane results in the Operator instances not being "recorded"
+by the Python function. What is more, there is actually *no composition logic*
+in between the ``QNode`` that is being created and the ``Operator`` instances
+(``qml.PauliX(0)`` and ``qml.PauliZ(0)``), as there are merely class
+instantiations happening upon calling the ``circuit`` quantum function.
+
+How does then PennyLane keep track of which ``Operations`` and ``Observables``
+were instantiated within a quantum function? The solution to this problem is
+related to the queuing of operators.
+
+Queuing of operators
+********************
+
+In PennyLane, the construction of quantum gates is separated from the specific
+quantum node (:class:`~.BaseQNode`) that they belong to. The reason for this is
+that, as mentioned before, the syntax of how ``Operator`` instances are
+included in quantum functions does not rely on or use ``QNode`` instances.
+Therefore, ``QNodes`` cannot keep track by default, which operators were
+included in the quantum function. A high-level object that holds information
+about the relationship between quantum gates and a quantum node can, however,
+help with this.
+
+The :class:`~.QueuingContext` class realizes this by providing access to the
+current QNode using the concept of a Python context manager. This is achieved
+by inheriting the ``BaseQNode`` class from ``QueuingContext``.  Furthermore, it
+provides the flexibility to have multiple objects record the creation of
+quantum gates.
+
+Details of queueing and specific operators can be described as follows:
+
+* ``Operators`` in a quantum function are queued on initialization.  This
+  happens via a call to :meth:`Operator.queue` which then interacts with the
```

Comment:
```suggestion
  happens via a call to :meth:`.Operator.queue` which then interacts with the parent
```

Reply:
The `QueuingContext` is not a parent to the `Operator` class, but a parent to `BaseQNode`